src/workflows-app/components/FilterableWorkflowTable.ts

```diff
@@ -151,6 +191,64 @@ const FilterableWorkflowTable = ({
     }
   };
 
+  const includeKey = useMemo(
+    () => [
+      'backendStatus',
+      'executionStatus',
+      'shardIndex',
+      'jobId',
+      'start',
+      'end',
+      'stderr',
+      'stdout',
+      'tes_stdout',
+      'tes_stderr',
+      'attempt',
+      'subWorkflowId', // needed for task type column
+    ],
+    []
+  );
+  const excludeKey = useMemo(() => [], []);
+
+  const getKeys = (keyValuePairs: {}): undefined => {
+    for (const key of Object.getOwnPropertyNames(keyValuePairs)) {
+      keyValuePairs[key];
+      delete Object.assign(keyValuePairs, {
+        [parseMethodString(key).variable]: keyValuePairs[key],
+      })[key];
+    }
+  };
+
+  const loadWorkflows = useCallback(
+    async (workflowId: string): Promise<WorkflowMetadata | void> => {
+      try {
+        const { cromwellProxyUrlState } = await loadAppUrls(workspaceId, 'cromwellProxyUrlState');
+        if (cromwellProxyUrlState.status === AppProxyUrlStatus.Ready) {
+          return await fetchMetadata(cromwellProxyUrlState.state, workflowId, signal, includeKey, excludeKey);
+        }
+      } catch (error) {
+        notify('error', 'Error loading run details', {
+          detail: error instanceof Response ? await error.text() : error,
+        });
+      }
+      return Promise.resolve();
+    },
+    [workspaceId, signal, includeKey, excludeKey]
+  );
+
+  /*
+   * Data fetchers
+   */
+  useEffect(() => {
+    const filteredPreviousRuns: Run[] = filterOption ? getFilteredRuns(filterOption, runsData) : runsData;
+    const firstPageIndex: number = (pageNumber - 1) * itemsPerPage;
+    const lastPageIndex: number = firstPageIndex + itemsPerPage;
+    const sortedRuns = sortRuns(sort.field, sort.direction, filteredPreviousRuns);
+    setSortedPreviousRuns(sortedRuns);
+    setPaginatedPreviousRuns(sortedRuns.slice(firstPageIndex, lastPageIndex));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [runsData, filterOption, pageNumber, itemsPerPage, sort.field, sort.direction]);
```

Comment:
This looks like a way to memoize `sortedPreviousRuns` and `paginatedPreviousRuns`. Consider whether these actually need to be stored in state or if they can be derived from state with `useMemo`.

Alternatively, `usePrevious` from libs/react-utils might be useful here.